Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +155,51 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            //For handling when array is null.
+            //If array is null then return number in array.
+            if (array == null)
+            {
+                return new [] { number };
+            }
+
+            //For handling when array is empty.
+            //If array is empty then there are two situations in test cases.
+            //One situation happens if also index is 0 then return number in array.
+            //Another situation happens if index is other than 0 then return empty array.
+            if (array.Length == 0)
+            {
+                if (index == 0)
+                {
+                    return new[] { number };
+                }
+                return new int[0];
+            }
+
+            var newArray = new int[array.Length + 1];
+            var j = 0;
+            for (var i = 0; i < newArray.Length; i++)
+            {
+                if (i == index)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4899409)